### PR TITLE
Document steps needed to create an Android Release app

### PR DESCRIPTION
**Pull request by mit-mit**
Initial docs for steps needed to prepare an app for release.

Fixes https://github.com/flutter/flutter/issues/6825

Fixes https://github.com/flutter/flutter/issues/7057

### CODE REVIEW

**Comment by mit-mit** [comment]
Thanks @LarkAscending for commits those great language edits!

**Comment by LarkAscending** [comment]
Yet another great contribution from @mit-mit! 
Very impressed by this article - it will really be helpful to users in the coming months :)

**Comment by eseidelGoogle** [comment]
Nifty!  Does this need to be linked from the rest of the site? :)

**Comment by mit-mit** [comment]
Yes, will add a link! What do we put in guides vs how-to's?

Also, does someone on the team have the knowledge to create a version of this for iOS?

**Comment by eseidelGoogle** [comment]
@cbracken is probably the right person to help with the iOS version.  He's currently helping a few folks in our gitter to go through the steps for iOS.

**Comment by LarkAscending** [comment]
@mit-mit What do you think of a new nav category, "Managing the App Lifecycle" ? or "Launching Your App"? 

Or?

File: android-release-build.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+1. Replace:
+```
+   android {
+```
+   with the keystore information from your properties file:
+```
+   def keystorePropertiesFile = rootProject.file("key.properties")
+   def keystoreProperties = new Properties()
+   keystoreProperties.load(new FileInputStream(keystorePropertiesFile))
```
Comment (jakobr-google) [inline android-release-build.md]: Tiny nit: This works, and is how most examples show how to do it, but it also leaves the properties file open until the build is done (or a GC occurs and the finalizers run). A version that closes the stream would look like:

```groovy
def keystoreProperties = new Properties()
rootProject.file('key.properties').withInputStream { stream ->
  keystoreProperties.load(stream)
}
```

But it's also fine to leave as is.

File: android-release-build.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+This section describes how to build a release APK. If you completed the 
+signing steps in the previous section, the release APK will be signed.
+
+Using the command line:
+
+1. `cd <app dir>` (replace `<app dir>` with your application's directory).
+1. Run `flutter build apk`.
+
+The release APK for your app is created in `<app dir>/android/app/build/outputs/app.apk`.
```
Comment (jakobr-google) [inline android-release-build.md]: We might want to use `<app dir>/android/app/build/outputs/apk/app-release.apk` here. If they do a `flutter run` afterwards, then `app.apk` will be overwritten, but `app-release.apk` will not.

File: android-release-build.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+at `<app dir>/android/src/main/AndroidManifest.xml`. Before releasing, 
+review the contents of this file and verify the values are correct, especially:
+
+* `application`: Edit the [`application`][applicationtag] tag to reflect the final name of the app.
+
+* `manifest`: Specify the [version][versions] code and name in the [`manifest`][manifesttag] tag.
+
+* `uses-sdk`: Specify the [API versions][apiversionstag] the app is compatible with.
+
+* `uses-permission`: Remove the `android.permission.INTERNET` [permission][permissiontag] if
```
Comment (mit-mit) [inline android-release-build.md]: Agree, opened https://github.com/flutter/flutter/issues/8421 to track this.

File: android-release-build.md (hunk 1 above)
Comment (mit-mit) [inline android-release-build.md]: I will leave it as it, as this is how all the official Android docs do it.

File: android-release-build.md (hunk 2 above)
Comment (mit-mit) [inline android-release-build.md]: Will fix!

**Comment by LarkAscending** [comment]
Seeing what might be a regression: for <li> elements, the line-height of 1.0 set in main.css is too low. In other words, bulleted and numbered list text is smooshed too close together.

If you Inspect in Chrome and deselect this property, the issue goes away.

**Comment by mit-mit** [comment]
@LarkAscending did you mean to type that comment in a different PR?

**Comment by LarkAscending** [comment]
No that's generally not something that I do...
I noticed the issue with this PR, and added a comment to take a look.

I then did a bunch of research on this side and saw it was site-wide, from
a PR in Jan by sfshaza.
Instead of deleting the comment here, in case you had seen it, I updated it
to say what was going on.

**Comment by mit-mit** [comment]
Ah, I understand, thanks!

**Pull request merged**